src/routes/auth/connexion.svelte

```diff
@@ -1,73 +1,96 @@
-<script>
-  import { onMount } from "svelte";
-  import { page } from "$app/stores";
+<script context="module">
+  import { token, setToken, validateCredsAndFillUserInfo } from "$lib/auth";
   import { goto } from "$app/navigation";
+  import { get } from "svelte/store";
+  import { getApiURL, defaultAcceptHeader } from "$lib/utils/api";
+
+  export async function load({ url, fetch }) {
+    function getNextPage() {
+      const next = url.searchParams.get("next");
+      if (next && next.startsWith("/") && !next.startsWith("/auth/"))
+        return next;
+      return "/";
+    }
 
-  import { getApiURL } from "$lib/utils/api.js";
-  import { token, setToken, validateCredsAndFillUserInfo } from "$lib/auth";
-  import { formErrors } from "$lib/validation.js";
-  import { loginSchema } from "$lib/schemas/auth.js";
-
-  import Button from "$lib/components/button.svelte";
-  import Fieldset from "$lib/components/forms/fieldset.svelte";
-  import Field from "$lib/components/forms/field.svelte";
-  import Alert from "$lib/components/forms/alert.svelte";
-  import Form from "$lib/components/forms/form.svelte";
+    // if we already have a token, bypass the page altogether
+    if (get(token)) {
+      goto(getNextPage());
+      // TODO: what should I return here?
+    }
+
+    const query = url.searchParams;
+    const code = query.get("code");
+    if (!code) {
+      // First arrival on the page
+      const targetUrl = `${getApiURL()}/inclusion-connect-login-info/`;
+      const result = await fetch(targetUrl, {
+        method: "GET",
+      });
+
+      if (result.ok) {
+        let { url: icRootUrl, state } = await result.json();
+        window.localStorage.setItem("oidcState", state);
+        const icUrl = `${icRootUrl}&state=${state}&redirect_uri=${url.href}`;
+        return {
+          props: {
+            icUrl,
+          },
+        };
+      }
+      // TODO return errors
+    } else {
+      // back from IC, with a code
+      const state = query.get("state");
+      const storedState = window.localStorage.getItem("oidcState");
+      window.localStorage.removeItem("oidcState");
+      if (!state || !storedState || state !== storedState) {
+        // TODO: raise error
+        // redirect to login page
+      }
+
+      const targetUrl = `${getApiURL()}/inclusion-connect-user-info/`;
```

Comment:
Au vu de son utilisation par le client, je pense que cet endpoint `/inclusion-connect-user-info` gagnerait à être renommé pour mieux refléter son utilité.

Ce que je constate qu'il fait :
 * recevoir `code` et `state` fournis par le provider OIDC ;
 * retourner un token d'authentification Dora.

Je ne connais pas vos conventions de nommage mais quelque chose qui s'approche de `/inclusion-connect-process-code` ou `/inclusion-connect-code-to-token` ou encore `/inclusion-connect-complete-login` me paraîtrait plus parlant ?